src/databricks/labs/ucx/workspace_access/generic.py

```diff
@@ -60,8 +60,17 @@ def _get_apply_task(
 
     @rate_limited(max_requests=30)
     def _applier_task(self, object_type: str, object_id: str, acl: list[iam.AccessControlRequest]):
-        self._ws.permissions.update(object_type, object_id, access_control_list=acl)
-        return True
+        permission = self._ws.permissions.update(object_type, object_id, access_control_list=acl)
+        request_acl = self.response_to_request(permission.access_control_list)
+
+        if acl == request_acl:
```

Comment:
1. comparing with response is not enough to detect configuration drift. it has to be a separate `get`.
2. plain `==` comparison does not work for it, because there might be other groups and/or users and/or service principals with permissions, that are not part of request. compare in a loop.
3. log a message - e.g. f'{object_type} {group} permissions {object_id} do not match: requested={requested} vs returned={returned}'

Reply:
Need to define the error message to make it easier, right now I feel I'm printing the entire group information for scim.py